### PR TITLE
Feat 2.3 scoring for 2.3.1, 2.3.2, 2.3.4

### PR DESCRIPTION
**Pull request by maxmalynowsky**
Adds scoring for geometry and column naming.

Misc changes:
- Run checks in parallel to increase runtime efficiency.

Other:
- Branched from: https://github.com/mapaction/cod-ab-data-quality/pull/12

### CODE REVIEW

File: src/checks/__main__.py [hunk 1]
```diff
@@ -91,12 +92,16 @@ def main() -> None:
             except DataSourceError:
                 gdf = GeoDataFrame()
             gdfs.append(gdf)
-        for function, results in checks:
-            rows = function.main(iso3, gdfs)
-            results.extend(rows)
+        with Pool() as pool:
```
Comment (Murray2015) [inline src/checks/__main__.py:95]: Nice, I think this is the right approach, as most of the runtime seems to be CPU bound.

File: src/scores/output.py [hunk 2]
```diff
@@ -51,19 +51,19 @@ def style(
         worksheet: Excel worksheet instance.
     """
     first_row = 1
-    first_col = 1
+    first_col = 2
```
Comment (Murray2015) [inline src/scores/output.py:54]: Where does this magic number come from?

File: src/scores/dates.py [hunk 3]
```diff
@@ -2,15 +2,15 @@
...
+        checks: checks DataFrame.
 
     Returns:
         Checks DataFrame with additional columns for scoring.
     """
-    df_score = df[["iso3", "level"]].copy()
-    df_score["dates"] = df["update_1"] > Timestamp.now() - relativedelta(years=3)
-    return df_score
+    scores = checks[["iso3", "level"]].copy()
+    scores["dates"] = checks["update_1"] > Timestamp.now() - relativedelta(years=1)
```
Comment (Murray2015) [inline src/scores/dates.py:15]: Where did the 3 / 1 year magic numbers come from?

File: src/scores/dates.py [hunk 4]
```diff
@@ -2,15 +2,15 @@
 from pandas import DataFrame, Timestamp
 
 
-def main(df: DataFrame) -> DataFrame:
+def main(checks: DataFrame) -> DataFrame:
     """Draft function for scoring date values within dataset.
```
Comment (Murray2015) [inline src/scores/dates.py:6]: Docstring might be out of date - is this still draft? Might be worth adding a TODO statement if it is so it's easier to find and fix in the future?

**Review by Murray2015 — COMMENTED** [review]
Only a few minor comments, all lgtm!

File: src/scores/dates.py (hunk 3 above)
Comment (maxmalynowsky) [reply src/scores/dates.py:15]: The old 3 years was a magic number I made up as a placeholder value. There is actually an annual COD review process where datasets should be validated on a 12 month cycle. (source: https://cod.unocha.org/)

Screenshot 2024-10-06 at 15 47 22

File: src/scores/dates.py (hunk 4 above)
Comment (maxmalynowsky) [reply src/scores/dates.py]: I put this in as a placeholder score from the beginning, I just updated `dates` and `languages` with all the scores that are possible to do at this point.

File: src/scores/output.py (hunk 2 above)
Comment (Murray2015) [reply src/scores/output.py]: Got it. Shall we add a comment or rename to make this clear?

File: src/scores/dates.py (hunk 3 above)
Comment (Murray2015) [reply src/scores/dates.py]: Thanks for explaining - I think that's worth adding into a comment, I don't think the next dev in 6 months will be able to guess this 🙂

**Review by Murray2015 — APPROVED** [review]

**Pull request merged**